### PR TITLE
Remove ByteBuffer pooling from the EphemeralFileSystemAbstraction

### PR DESCRIPTION
**Pull request by chrisvest**
This will hopefully make it less likely that builds fail because they run out of direct memory.

### CODE REVIEW

File: community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java [hunk 1]
```diff
@@ -1131,33 +1089,24 @@ private void copyByteBufferContents( ByteBuffer from, ByteBuffer to )
...
-                                return buffer;
-                            }
-                        }
-                    }
+                    return ByteBuffer.allocate( capacity );
+                }
+                catch ( OutOfMemoryError secondOom )
+                {
+                    secondOom.addSuppressed( oom );
+                    throw secondOom;
```
Comment (thobe) [inline community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java]: This is backwards, typically you would throw the original exception, and have the outer exception suppress the inner exception. Here you suppress the outer exception with the inner one, which doesn't seem natural.

**Comment by thobe** [comment]
Nice and simple change, I like how it simplifies the code. If it does what it says on the package that'll be awesome!

File: community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java (hunk 1 above)
Comment (chrisvest) [inline community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java]: Ah, okay. I'll flip them around.

**Comment by mneedham** [comment]
Tried out a personal build on TeamCity with the failing StoreMigratorTest pointing at Ephemeral FS and it worked fine. Hoorah!

**Pull request merged**